src/main/java/eu/decentsoftware/holograms/api/utils/items/HologramItem.java

```diff
@@ -95,15 +145,30 @@ private void applyNBT(Player player, ItemStack itemStack) {
     private void parseContent() {
         String string = this.content;
         string = findExtras(string);
-        string = findNBT(string);
+        //string = findNBT(string);
         string = checkEnchanted(string);
+        string = findModelData(string);
         parseMaterial(string);
 
         if (this.material == null) {
             this.material = Material.STONE;
         }
     }
 
+    private String findModelData(String string) {
+        String prefix = "{CustomModelData:";
+        int start = string.indexOf(prefix);
+        int end = string.indexOf('}', start);
+
+        if (start != -1 && end != -1) {
+            String modelDataString = string.substring(start + prefix.length(), end);
+            this.modelData = Integer.parseInt(modelDataString);
+            string = string.substring(0, start) + string.substring(end + 1);
+        }
+
+        return string;
+    }
```

Comment:
This is not a good aproach for various reasons. The main one being that you effectively assume that there is only `{CustomModelData:...}` itself, which is just wrong.

First of is the NBT not always just the CustomModelData itself, but a collection of various ones, like for example a color value for a dyable item, which also provides a second issue, which is the fact that checking for `}` is not a reliable way of doing things here, given there can be more info before/after the entire CMD.

To give an example, imagine the following text:
```
#ICON:POTION {display:{color:1234567},CustomModelData:10000}
```
This one would already fail to find the CMD because it doesn't starts with `{CustomModelData:`. And even if it would be adjusted to have CMD be at the start (`{CustomModelData:10000,display:{color:1234567}}`) would the resulting CMD value be `10000,display:{color:1234567`.

Next, calling `Integer.parseInt` like that is not safe to do, as this method will throw an exception in case of a non-number being provided, so it should be at the very least be put in a try-catch to ensure it doesn't cause issues.